actix-http/src/encoding/decoder.rs

```diff
@@ -79,15 +79,8 @@ where
     ) -> Poll<Option<Self::Item>> {
         loop {
             if let Some(ref mut fut) = self.fut {
-                let (chunk, decoder) = match ready!(Pin::new(fut).poll(cx)) {
-                    Ok(Ok(item)) => item,
-                    Ok(Err(e)) => {
-                        return Poll::Ready(Some(Err(BlockingError::Error(e).into())))
-                    }
-                    Err(_) => {
-                        return Poll::Ready(Some(Err(BlockingError::Canceled.into())))
-                    }
-                };
+                let (chunk, decoder) =
+                    ready!(Pin::new(fut).poll(cx)).map_err(|_| BlockingError)??;
```

Comment:
That seems will clear error, which might be fs error too.. odds are low but might complicate debugging I guess

Reply:
the fs error will be propagated by the second ?

Reply:
`std::io::Error` would be converted to `PayloadError::Incomplete variant`.
The old behavior is to wrap io::Error in BlockingError::Error and then extract it as PayloadError::Io variant.
It's kinda an inconsistent as all other io::Error are treated as Incomplete variant.

Reply:
Yeah I got confused, sorry